flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -155,13 +156,14 @@ public void startCluster() throws ClusterEntrypointException {
 		LOG.info("Starting {}.", getClass().getSimpleName());
 
 		try {
-
-			configureFileSystems(configuration);
+			//TODO: push down filesystem initialization into runCluster - initializeServices (?)
```

Comment:
It is sort of the main question that I had - if it is OK to load FileSystems not there they are loaded now, but at the later state, for instance inside initilizeServices.

Reply:
I don't know if it is ok to move it or not, it might be risky. Two potential issues:
- `SecurityContext` - does it affect initialisation in anyway?
- We have to be sure that we initialise `FileSystems` before we are going to need them. This might be relevant if there are services that depend on `FileSystems` - if that's the case, we would have to take that into account (correctly the initialisation)

Apart from that, it might create some unknown untested side effects that I'm simply not aware of, so I would weakly vote on leaving it as it is, unless this is causing some problems for us?

Reply:
Just mild irritation of some "spaghetti"-like initialization process and increased API ugliness because of leaving the process as is. I am fine with that for now, but it seems like some refactoring might be due at a later point.